app/Http/Controllers/Auth/OAuthController.php

```diff
@@ -44,47 +44,97 @@ public function redirect($provider)
      * @param  string  $driver
      * @return \Illuminate\Http\Response
      */
-    public function handleCallback($provider)
+    public function handleCallback(OAuthProviderService $provider)
     {
-        $user = Socialite::driver($provider)->stateless()->user();
-        $user = $this->findOrCreateUser($provider, $user);
-
-        $this->guard()->setToken(
-            $token = $this->guard()->login($user)
-        );
-
-        return view('oauth/callback', [
-            'token' => $token,
-            'token_type' => 'bearer',
-            'expires_in' => $this->guard()->getPayload()->get('exp') - time(),
-        ]);
+        try {
```

Comment:
The try catch doesn't need to be around everything - you can just put ut around the first line I believe? 
There should not be exception otherwise?